smartsim/_core/launcher/colocated.py

```diff
@@ -100,6 +99,10 @@ def _build_colocated_wrapper_cmd(
     :type port: int
     :param ifname: network interface(s) to bind DB to
     :type ifname: str | list[str], optional
+    :param limit_db_cpus: If True, limit the cpus that the database can run on
+    :type limit_db_cpus: bool, optional
```

Comment:
Can we remove this parameter and just assume that if a user provides a `db_cpu_list` that is not None that they want to limit to those cpus?

Reply:
Not too keen on removing `limit_db_cpus` as I think there's enough people who are ok with the 'automatic' creation of `db_cpu_list`. What I'd prefer to do is to raise an exception if `limit_db_cpus` is False but `db_cpu_list` isn't None. It's potentially ambiguous what the user wants to do.

Reply:
I can see removing it in this routine though since this shouldn't be touched by the user.